crates/holochain_conductor_api/src/app_interface.rs

```diff
@@ -254,70 +270,51 @@ pub enum AppResponse {
     Ok,
 }
 
-/// The data provided over an app interface in order to make a zome call
+/// The data provided over an app interface in order to make a zome call.
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct ZomeCall {
-    /// The ID of the cell containing the zome to be called
-    pub cell_id: CellId,
-    /// The zome containing the function to be called
-    pub zome_name: ZomeName,
-    /// The name of the zome function to call
-    pub fn_name: FunctionName,
-    /// The serialized data to pass as an argument to the zome function call
-    pub payload: ExternIO,
-    /// The capability request authorization
-    ///
-    /// This can be `None` and still succeed in the case where the function
-    /// in the zome being called has been given an `Unrestricted` status
-    /// via a `CapGrant`. Otherwise it will be necessary to provide a `CapSecret` for every call.
-    pub cap_secret: Option<CapSecret>,
-    /// The provenance (source) of the call
-    /// MUST match the signature.
-    pub provenance: AgentPubKey,
+pub struct ZomeCallParamsSigned {
+    /// Bytes of the serialized zome call payload that consists of all fields of the
+    /// [`ZomeCallParams`].
+    pub bytes: ExternIO,
+    /// Signature by the provenance of the call, signing the bytes of the zome call payload.
     pub signature: Signature,
-    pub nonce: Nonce256Bits,
-    pub expires_at: Timestamp,
 }
 
-impl From<ZomeCall> for ZomeCallUnsigned {
-    fn from(zome_call: ZomeCall) -> Self {
+impl ZomeCallParamsSigned {
+    pub fn new(bytes: Arc<[u8]>, signature: Signature) -> Self {
         Self {
-            cell_id: zome_call.cell_id,
-            zome_name: zome_call.zome_name,
-            fn_name: zome_call.fn_name,
-            payload: zome_call.payload,
-            cap_secret: zome_call.cap_secret,
-            provenance: zome_call.provenance,
-            nonce: zome_call.nonce,
-            expires_at: zome_call.expires_at,
+            bytes: ExternIO::from(bytes.to_vec()),
+            signature,
         }
     }
+
+    pub async fn try_from_params(
+        keystore: &MetaLairClient,
+        params: ZomeCallParams,
+    ) -> LairResult<Self> {
+        let bytes = params.data_to_sign().map_err(|e| e.to_string())?;
+        let signature = params.provenance.sign_raw(keystore, bytes.clone()).await?;
+        Ok(Self::new(bytes, signature))
+    }
+}
+
+///
+#[derive(Clone, Debug)]
+pub struct ZomeCall {
```

Comment:
This type is used internally and consists of the params (deserialized) and the signed part (bytes + signature).